docs/docs/the-component-lifecycle.md

```diff
@@ -0,0 +1,43 @@
+---
+id: the-component-lifecycle
+title: The Component Lifecycle
+permalink: docs/the-component-lifecycle.html
+prev: rendering-elements.html
+---
+
+Components have three main parts of their lifecycle:
+
+* **[Mounting](#mounting):** A component is being inserted into the DOM.
+* **[Updating](#updating):** A component is being re-rendered to determine if the DOM should be updated.
+* **[Unmounting](#unmounting):** A component is being removed from the DOM.
+
+React provides lifecycle methods that you can specify to hook into this process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+## Mounting
+
+These methods are called when a component is being inserted into the DOM.
+
+* **`getInitialState(): object`** is invoked before a component is mounted. Stateful components should implement this and return the initial state data.
+* **`componentWillMount()`** is invoked immediately before mounting occurs.
+* **`componentDidMount()`** is invoked immediately after mounting occurs. Initialization that requires DOM nodes should go here.
+
+## Updating
+
+These methods are called when a component is being re-rendered and are used to determine if the DOM should be updated.
+
+* **`componentWillReceiveProps(object nextProps)`** is invoked when a mounted component receives new props. This method should be used to compare `this.props` and `nextProps` to perform state transitions using `this.setState()`.
```

Comment:
Can we clarify it occurs *before* it receives new props?

Also, "should be used" makes it sound like you *need* to do that but it's actually a rare (and slightly discouraged because of complexity) case.

Can reword as:

"If you need to update the state in response to prop changes (for example, to reset it), you may compare `this.props` and `nextProps` and perform state transitions using `this.setState()` in this method. Note that React may call this method even if the props have not changed, so make sure to compare the previous and current values if you only want to handle changes."